ledger/query/src/query/rest.rs

```diff
@@ -0,0 +1,319 @@
+    }
+
+    /// Returns a state path for the given `commitment`.
+    fn get_state_path_for_commitment(&self, commitment: &Field<N>) -> Result<StatePath<N>> {
+        self.get_request(&format!("statePath/{commitment}"))
+    }
+
+    /// Returns a state path for the given `commitment`.
+    #[cfg(feature = "async")]
+    async fn get_state_path_for_commitment_async(&self, commitment: &Field<N>) -> Result<StatePath<N>> {
+        self.get_request_async(&format!("statePath/{commitment}")).await
+    }
+
+    /// Returns a list of state paths for the given list of `commitment`s.
+    fn get_state_paths_for_commitments(&self, commitments: &[Field<N>]) -> Result<Vec<StatePath<N>>> {
+        // Construct the comma separated string of commitments.
+        let commitments_string = commitments.iter().map(|cm| cm.to_string()).collect::<Vec<_>>().join(",");
+        self.get_request(&format!("statePaths?commitments={commitments_string}"))
+    }
+
+    /// Returns a list of state paths for the given list of `commitment`s.
+    #[cfg(feature = "async")]
+    async fn get_state_paths_for_commitments_async(&self, commitments: &[Field<N>]) -> Result<Vec<StatePath<N>>> {
+        // Construct the comma separated string of commitments.
+        let commitments_string = commitments.iter().map(|cm| cm.to_string()).collect::<Vec<_>>().join(",");
+        self.get_request_async(&format!("statePaths?commitments={commitments_string}")).await
+    }
+
+    /// Returns a state path for the given `commitment`.
+    fn current_block_height(&self) -> Result<u32> {
+        self.get_request("block/height/latest")
+    }
+
+    /// Returns a state path for the given `commitment`.
+    #[cfg(feature = "async")]
+    async fn current_block_height_async(&self) -> Result<u32> {
+        self.get_request_async("block/height/latest").await
+    }
+}
+
+impl<N: Network> RestQuery<N> {
+    /// The API version to use when querying REST endpoints.
+    const API_VERSION: &str = "v2";
+
+    /// Returns the transaction for the given transaction ID.
+    pub fn get_transaction(&self, transaction_id: &N::TransactionID) -> Result<Transaction<N>> {
+        self.get_request(&format!("ransaction/{transaction_id}"))
+    }
+
+    /// Returns the transaction for the given transaction ID.
+    #[cfg(feature = "async")]
+    pub async fn get_transaction_async(&self, transaction_id: &N::TransactionID) -> Result<Transaction<N>> {
+        self.get_request_async(&format!("transaction/{transaction_id}")).await
+    }
+
+    /// Returns the program for the given program ID.
+    pub fn get_program(&self, program_id: &ProgramID<N>) -> Result<Program<N>> {
+        self.get_request(&format!("program/{program_id}"))
+    }
+
+    /// Returns the program for the given program ID.
+    #[cfg(feature = "async")]
+    pub async fn get_program_async(&self, program_id: &ProgramID<N>) -> Result<Program<N>> {
+        self.get_request_async(&format!("program/{program_id}")).await
+    }
```

Comment:
I think I noted in a previous PR that I don't see why we should replicate a subset of our API calls in Query. Moreover, this code contains a critical typo, which underscores the need to minimize our maintainable codebase. :)